connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -143,6 +163,90 @@ public ConnectorStateInfo connectorStatus(String connName) {
                 status.workerId(), status.trace());
     }
 
+
+    @Override
+    public ConfigInfos validateConfigs(String connType, Map<String, String> connectorConfig) {
+        ConfigDef connectorConfigDef = ConnectorConfig.configDef();
+        List<ConfigValue> connectorConfigValues = connectorConfigDef.validate(connectorConfig);
+        ConfigInfos result = generateResult(connType, connectorConfigDef.configKeys(), connectorConfigValues, Collections.<String>emptyList());
+
+        if (result.errorCount() != 0) {
+            return result;
+        }
+
+        ConnectorConfig connConfig = new ConnectorConfig(connectorConfig);
+        String connName = connConfig.getString(ConnectorConfig.NAME_CONFIG);
+        Connector connector = getConnector(connType, connName);
+
+        Config config = connector.validate(connectorConfig);
+
+        Map<String, ConfigKey> configKeys = config.configDef().configKeys();
+        List<ConfigValue> configValues = config.configValues();
+
+        Map<String, ConfigKey> resultConfigKeys = new HashMap<>(configKeys);
+        resultConfigKeys.putAll(connectorConfigDef.configKeys());
```

Comment:
In this round of implementation, I removed the `configDef()` getter in the connector class and instead add that in the return of `validate()`. Probably better to add `configDef()` back or reuse `defineConfig()`?